### DISCUSSION TITLE
Flat style geometry expression

### DISCUSSION THREAD

**Discussion by gberaudo** [openlayers/openlayers#15545] [Ideas]
With @jahow  and @fredj  we have been thinking about adding a geometry expression to the flat styles.

For example, to display arrows along a line, with a density depending on zoom level, we may write:

```js
{
  stroke-width: 2,
  stroke-color: ‘#a44’
},
{
  icon-src: ‘spritesheet.svg’,
  icon-offset: [128, 16],
  icon-size: [16, 16],
  icon-rotation: [‘+’, [‘get’, ‘$orientation’], 90],
  filter: [‘geometry-type’, ‘LineString’],
  geometry: [‘point-along-line’, [‘*’, [‘zoom’], 0.1]]
}
```

Have others already thought/experimented about it?
Are you interested in such feature?
Looking forward for your inputs.

### Point along line
`[‘point-along-line’, <interval in px>, <offset in px>]`
> $orientation

### Interior point
`[‘interior-point’]`

### Vertices
`[‘vertices’]`
> $currentIndex
> $vertexCount
> $orientationPrev
> $orientationNext

### Simplify
`['simplify', <factor>]`

**Comment by jahow** [comment]
Some more options to take into consideration:

```js
  {
      'icon-src': 'icon.svg',
      geometry: ['line-start', 12], // offset in pixel
      geometry: ['line-end', 12], // offset in pixel
      geometry: ['line-center'],
      geometry: ['along-line', '%', 0.25], // repeated every 25%
      geometry: ['along-line', 'px', 15], // repeated every 15px
  }
```
